### PR TITLE
Fixed bugs in list-services update

### PR DESCRIPTION
**Pull request by npasricha-cs**
Creating pull request to add onto ticket for testing before merge.

Ticket: https://cognitivescale.atlassian.net/browse/CCF-9802

Fixed the bugs with previous update which could mess up when '$ref' is the parameter type rather than a list of actual params.

### CODE REVIEW

**Comment by mburpo-CS** [comment]
This is the PR for https://cognitivescale.atlassian.net/browse/CCF-9802

**Comment by vadan-cs** [comment]
Can we add tests for this?

**Comment by daldridge-cs** [comment]
This looks like a re-introduction of a change that was explicitly reverted last week.  How is this change different than what was attempted then?

**Comment by daldridge-cs** [comment]
> Don't include (remove) `.DS_Store` files

Please add to `.gitignore`

**Comment by npasricha-cs** [comment]
Testing was not done for what was introduced last week, and some bugs were found. The changes were reverted as a result. Fixes for the issues last week have been added, but I think adding some tests for this would be the next step.

**Comment by cdraper-CS** [comment]
Code needs to deal with possibility of no parameters element. Suggest use lodash get.

File: src/client/catalog.js [hunk 1]
```diff
@@ -126,6 +126,18 @@ module.exports = class Catalog {
                 const servicesList = response.agent.inputs
                     .filter(i => i.signalType === 'Service')
                     .map(i => ({ ...i, url: `${urlBase}/${i.name}` }))
+                    .map(i => ({ ...i, formated_types:
+                    (i.parameters === null ? null
```
Comment (daldridge-cs) [inline src/client/catalog.js]: Yikes, nested ternary, with an embedded map operation in the last operand?!?  Makes my brain hurt to read it.  I'd suggest moving the `o => ...` function in the final map (excluding the join) to some utility method, maybe `formatServiceInputParameter(...)`, and moving the final/nested ternary (detecting $ref or not, and joining non-ref results) to a utility method like `formatAllServiceInputParameters(...)`.  I dunno, just thinking/typing out loud.  Definitely needs to be more readable.

Comment (daldridge-cs) [reply src/client/catalog.js]: Also, typos.  `formated` -> `formatted`

Comment (npasricha-cs) [reply src/client/catalog.js]: I wasn't sure if I was allowed to make my own custom functions to be honest, but that is a great idea so I did it. I just pushed the changed. It looks a lot cleaner now, thanks for the suggestion.

File: .gitignore [hunk 2]
```diff
@@ -96,3 +96,5 @@ test/cortex/local.json
 .local/
 
 dist/
+
+package-lock.json
```
Comment (daldridge-cs) [inline .gitignore]: I've been committing (not ignoring) package-lock.json on my other projects (Studio, etc.).  The file is required to exists to be able to run `npm audit`, for example.  Have others on the team indicated it should be ignore/omitted for the CLI?  I'm interested in the rationale.

Comment (npasricha-cs) [reply .gitignore]: So I am not exactly sure, but I saw that the .gitignore file on develop has it listed there so I put it back in mine

Comment (daldridge-cs) [reply .gitignore]: This may have come up on Slack recently.  I think Johan had an opinion?  Talk with him, please.

Comment (jgielstra-cs) [reply .gitignore]: We've been leaving it out as each time someone runs npm install it gets updated.
Also if developers/CI have slightly different settings in .npmrc it breaks the build.
So you get inadvertent updates, stability issues, false sense of security ( see npm-shrinkwrap.json ) 

Just done use ^ or ~

Comment (npasricha-cs) [reply .gitignore]: Ok I left it out. Thanks for the help

**Pull request merged**